src/main/flight/pid_init.c

```diff
@@ -267,6 +267,50 @@ void pidInitFilters(const pidProfile_t *pidProfile)
 #endif
 }
 
+
+#ifdef USE_ADVANCED_TPA
+float tpaCurveHyperbolicFunction(float x, void *args)
+{
+    const pidProfile_t *pidProfile = (const pidProfile_t*)args;
+
+    const float thrStall = pidProfile->tpa_curve_stall_throttle / 100.0f;
+    const float pidThr0 = pidProfile->tpa_curve_pid_thr0 / 100.0f;
+
+    if (x <= thrStall) {
+        return pidThr0;
+    }
+
+
+    const float expoDivider = pidProfile->tpa_curve_expo / 10.0f - 1.0f;
+    const float expo = (fabsf(expoDivider) > 1e-3f) ?  1.0f / expoDivider : 1e3f; // avoiding division by zero for const float base = ...
+
+    const float pidThr100 = pidProfile->tpa_curve_pid_thr100 / 100.0f;
+    const float xShifted = scaleRangef(x, thrStall, 1.0f, 0.0f, 1.0f);
+    const float base = (1 + (powf(pidThr0 / pidThr100, 1.0f / expo) - 1) * xShifted);
+    const float divisor = powf(base, expo);
+
+    return pidThr0 / divisor;
+}
+
+void tpaCurveHyperbolicInit(const pidProfile_t *pidProfile)
+{
+    pwlInitialize(&pidRuntime.tpaCurvePwl, pidRuntime.tpaCurvePwl_yValues, TPA_CURVE_PWL_SIZE, 0.0f, 1.0f);
+    pwlFill(&pidRuntime.tpaCurvePwl, tpaCurveHyperbolicFunction, (void*)pidProfile);
+}
+
+void tpaCurveInit(const pidProfile_t *pidProfile)
+{
+        switch (pidProfile->tpa_curve_type) {
+        case TPA_CURVE_HYPERBOLIC:
+            tpaCurveHyperbolicInit(pidProfile);
+            return;
+        case TPA_CURVE_CLASSIC:
+        default:
+            return;
+        }
```

Comment:
```suggestion
        if (pidProfile->tpa_curve_type == TPA_CURVE_HYPERBOLIC) {
            tpaCurveHyperbolicInit(pidProfile);
        }
```

Reply:
left switch here for future improvements for CLASSIC and maybe other types.